net/can/can_recvmsg.c

```diff
@@ -410,8 +410,7 @@ static uint16_t can_recvfrom_eventhandler(FAR struct net_driver_s *dev,
                                           FAR void *pvpriv, uint16_t flags)
 {
   struct can_recvfrom_s *pstate = (struct can_recvfrom_s *)pvpriv;
-#if (defined(CONFIG_NET_CANPROTO_OPTIONS) && defined(CONFIG_NET_CAN_CANFD)) || \
-  defined(CONFIG_NET_TIMESTAMP)
```

Comment:
This'll cause CI fail for b-g431b-esc1/cansock configuration:

> Error: can/can_recvmsg.c:415:22: error: unused variable 'conn' [-Werror=unused-variable]
>   415 |   struct can_conn_s *conn = (struct can_conn_s *)pstate->pr_sock->s_conn;
>       |                      ^~~~

Probably we should modify this condition to just:
`#if defined(CONFIG_NET_CANPROTO_OPTIONS) || defined(CONFIG_NET_TIMESTAMP)
`

Reply:
Sorry, which line are you referring to as causing the current error?  Line 426?

Reply:
This is not related to the current error. CI didn't take this stage yet (arm-07)
You can look at https://github.com/apache/incubator-nuttx/runs/5172919938?check_suite_focus=true#step:7:124 which was part of PR https://github.com/apache/incubator-nuttx/pull/5481.
To fix CI I changed these lines with https://github.com/apache/incubator-nuttx/pull/5481/commits/e86e6b841f508aea6826d0554aef921f6cf077e0 but I see now it was not correct.

Reply:
I believe my changes fix that error, correct?  I don't see any combination of CONFIG's that would result in either an undefined variable or an unused variable.

Reply:
Yes, it is correct. Sorry, my bad, I somehow saw it wrong on diff.